hedera-node/src/main/java/com/hedera/services/txns/file/FileDeleteTransitionLogic.java

```diff
@@ -56,31 +58,19 @@ public FileDeleteTransitionLogic(HederaFs hfs, TransactionContext txnCtx) {
 
 	@Override
 	public void doStateTransition() {
-		var op = txnCtx.accessor().getTxn().getFileDelete();
-
-		try {
-			var tbd = op.getFileID();
-			if (!hfs.exists(tbd)) {
-				txnCtx.setStatus(INVALID_FILE_ID);
-				return;
-			}
-
-			var attr = hfs.getattr(tbd);
-			if (attr.getWacl().isEmpty()) {
-				txnCtx.setStatus(UNAUTHORIZED);
-				return;
-			}
-			if (attr.isDeleted()) {
-				txnCtx.setStatus(FILE_DELETED);
-				return;
-			}
-
-			var result = hfs.delete(tbd);
-			txnCtx.setStatus(result.outcome());
-		} catch (Exception unknown) {
-			log.warn("Unrecognized failure handling {}!", txnCtx.accessor().getSignedTxnWrapper(), unknown);
-			txnCtx.setStatus(FAIL_INVALID);
-		}
+		/* --- Extract from gRPC --- */
+		final var op = txnCtx.accessor().getTxn().getFileDelete();
+		final var tbd = op.getFileID();
+		
+		/* --- Perform validations --- */
+		validateTrue(hfs.exists(tbd), INVALID_FILE_ID);
+		final var attr = hfs.getattr(tbd);
+		validateFalse(attr.getWacl().isEmpty(), UNAUTHORIZED);
+		validateFalse(attr.isDeleted(), FILE_DELETED);
+		
+		/* --- Do the business logic --- */
+		final var result = hfs.delete(tbd);
+		validateTrue(result.outcome() == SUCCESS, result.outcome());
```

Comment:
Would it be possible to refactor `hfs.delete` to throw instead of return `ResponseCode`?